parlai/agents/transformer/modules.py

```diff
@@ -956,41 +1024,43 @@ def forward(self, x, encoder_output, encoder_mask, incr_state=None):
         if self.variant == 'aiayn' or self.variant == 'xlm' or self.variant == 'bart':
             x = self.norm3(x)
 
-        new_incr_state = {
-            'self_attn': final_self_attn_incr_state,
-            'encoder_attn': final_encoder_attn_incr_state,
-        }
+        new_incr_state = {}
+        for type_ in attn_types:
+            new_incr_state.update(
+                {
+                    f'self_attn_{type_}': final_self_attn_incr_state[type_],
+                    f'encoder_attn_{type_}': final_encoder_attn_incr_state[type_],
+                }
+            )
         return x, new_incr_state
 
     def _create_selfattn_mask(self, x):
         # figure out how many timestamps we need
         bsz = x.size(0)
         time = x.size(1)
         # make sure that we don't look into the future
-        mask = torch.tril(x.new(time, time).fill_(1))
+        mask = torch.tril(torch.ones(time, time, dtype=x.dtype, device=x.device))
         # broadcast across batch
         mask = mask.unsqueeze(0).expand(bsz, -1, -1)
         return mask
 
-    def reorder_incremental_state(
-        self, incremental_state: Dict[str, dict], inds: torch.Tensor
-    ) -> Dict[str, dict]:
+
+class NegInfMixin:
```

Comment:
Sorry, I don't understand why this is a mixin.

Reply:
It feels obtuse. What other options do we have?

Reply:
Okay, was able to revert this - this might be something that wasn't supported when scripting but is when tracing, I suspect